.azure-pipelines/ultimate-pipeline.yml

```
@@ -4989,107 +4989,163 @@ stages:
-          WEBLOG_VARIANT: "poc"
-          SCENARIO: DEFAULT
-        default_uds:
-          WEBLOG_VARIANT: "uds"
-          SCENARIO: DEFAULT
-        integrations_poc:
-          WEBLOG_VARIANT: "poc"
-          SCENARIO: INTEGRATIONS
-        integrations_uds:
-          WEBLOG_VARIANT: "uds"
-          SCENARIO: INTEGRATIONS
-        integrations_aws_poc:
-          WEBLOG_VARIANT: "poc"
-          SCENARIO: INTEGRATIONS_AWS
-        integrations_aws_uds:
-          WEBLOG_VARIANT: "uds"
-          SCENARIO: INTEGRATIONS_AWS
-        crossed_tracing_libraries_poc:
-          WEBLOG_VARIANT: "poc"
-          SCENARIO: CROSSED_TRACING_LIBRARIES
-        crossed_tracing_libraries_uds:
-          WEBLOG_VARIANT: "uds"
-          SCENARIO: CROSSED_TRACING_LIBRARIES
-        # All the other scenarios that aren't in a secenario group
-        remainder_poc:
-          WEBLOG_VARIANT: "poc"
-          SCENARIO: "+S TRACE_PROPAGATION_STYLE_W3C +S PROFILING +S LIBRARY_CONF_CUSTOM_HEADERS_SHORT +S LIBRARY_CONF_CUSTOM_HEADERS_LONG"
-        remainder_uds:
-          WEBLOG_VARIANT: "uds"
-          SCENARIO: "+S TRACE_PROPAGATION_STYLE_W3C +S PROFILING +S LIBRARY_CONF_CUSTOM_HEADERS_SHORT +S LIBRARY_CONF_CUSTOM_HEADERS_LONG"
-        debugger_poc:
-          WEBLOG_VARIANT: "poc"
-          SCENARIO: DEBUGGER_SCENARIOS
-        debugger_uds:
-          WEBLOG_VARIANT: "uds"
-          SCENARIO: DEBUGGER_SCENARIOS
+        - script: |
+            PACKAGE_NAME=$(basename $(Build.ArtifactStagingDirectory)/datadog-dotnet-apm-*.tar.gz)
+            echo Moving $PACKAGE_NAME to system-tests/binaries
+            mv $(Build.ArtifactStagingDirectory)/$PACKAGE_NAME system-tests/binaries/
+          displayName: Move dotnet binary to system test folder
+          condition: eq( variables['System.JobAttempt'], 1 )
+        
+        - script: |
+            set -e
+            cd system-tests
+            mkdir -p ../docker-images
+        
+            while read w; do
+              echo "------------------------------------"
+              echo "Building images for variant: $w"
+              echo "------------------------------------"
+              # Build the images. For example:
+              ./build.sh dotnet --weblog-variant $w
+            
+              # Save weblog variant image
+              docker save system_tests/weblog:latest -o ../docker-images/$w.tar
+            done < /tmp/weblogs_list.txt
+            
+            # Save agent image
+            docker save system_tests/agent:latest -o ../docker-images/agent.tar
 
-    steps:
+          displayName: Build weblogs and agent & save Docker images
+          condition: eq( variables['System.JobAttempt'], 1 )
+        
+        - task: PublishPipelineArtifact@1
+          displayName: Publish Docker images artifact
+          condition: eq( variables['System.JobAttempt'], 1 )
+          inputs:
+            path: 'docker-images'
+            artifactName: 'docker-images'
```

could we rename this artifact so it makes more sense in the big list? e.g.
```suggestion
            artifactName: 'system-test-docker-images'
```